table/block_based/block_based_table_reader.cc

```diff
@@ -1633,6 +1637,9 @@ void BlockBasedTable::RetrieveMultipleBlocks(
     // If current block is adjacent to the previous one, at the same time,
     // compression is enabled and there is no compressed cache, we combine
     // the two block read as one.
+    // We don't combine block reads here in direct IO mode, because when doing
+    // direct IO read, the block requests will be realigned and merged when
+    // necessary.
     if (scratch != nullptr && prev_end == handle.offset()) {
```

Comment:
Maybe its just me, but I think using ```use_shared_buffer``` here and below and then differentiating using ```file->use_direct_io()``` might be easier to reason about, instead of checking ```scratch```.

Reply:
make sense. The check on `scratch` contains implicit assumption that if `use_direct_io`, then `scratch` will be set to `nullptr` in `MultiGet`.